Automattic/sensei
# Learning mode - Change all question styles for quiz page

Resolves https://github.com/Automattic/sensei/issues/7070

Stacked on https://github.com/Automattic/sensei/pull/7106

This PR contains the styles quiz 'taking' page. As we're working on the LM designs of quiz, not every change of design is applied in the editor, as many CSS styles are applied at runtime overriding the current theme's styles for LM. But the editor styles have been updated too for most font stylings and spacings.

We'll handle the quiz 'submitted' page in another PR.

## Proposed Changes
* Makes all the quiz, question and answer related blocks accept styles via theme.json and variations
* Adds styles for File Upload input
* Changes Checkbox styles
* Changes Radio button styles
* Adds code to allow adding CSS inside tinyMCE editor's content area inside iframe
* The tinyMCE version Wordpress provides (4.6.2) doesn't have placeholder (starts from 5.2), so we've implemented that with custom js.
* Overall updates all the styles for each question to match the styles as closely as possible for all variations of Course theme and other themes. Updated Non-LM styles so that it doesn't look broken either.

## Testing Instructions


There are many combinations here. 7 questions (With and without input) * 4 variations (For course theme only) * 2 states (LM and Non - LM) * different screen sizes. All the states are not matched pixel perfectly, we tried to utilize existing spacings and font sizes wherever possible. So it shouldn't look too much different overall.

1. Check out this branch of Course theme https://github.com/Automattic/themes/pull/7331
2. Check out this branch of Pro https://github.com/Automattic/sensei-pro/pull/2403
3. Enable course theme
4. Create a course with lessons
5. Have at least one lesson with a quiz that has all the questions in it
6. Make sure Learning Mode is enabled for that course
7. Now check that quiz as a student
8. Check in mobile screen size as well
9. Change the theme variation to gold, dark and blue and check the styles for them too.
10. Now switch to another block theme and check
11. Switch to a non-block theme and check
12. Check the editor and make sure it doesn't look broken.
13. For all themes and variations, also check with LM disabled to make sure non-LM style isn't broken

![image](https://github.com/Automattic/sensei/assets/6820724/39852395-d749-4f54-a896-508aa23a23e0)

![image](https://github.com/Automattic/sensei/assets/6820724/e85a4da7-cfc7-4a1a-a70b-3df3895a9aeb)


## Pre-Merge Checklist
